go/tasks/plugins/webapi/bigquery/query_job.go

```diff
@@ -0,0 +1,259 @@
+
+	// MaximumBytesBilled: [Optional] Limits the bytes billed for this job.
+	// Queries that will have bytes billed beyond this limit will fail
+	// (without incurring a charge). If unspecified, this will be set to
+	// your project default.
+	MaximumBytesBilled int64 `json:"maximumBytesBilled,omitempty,string"`
+
+	// Priority: [Optional] Specifies a priority for the query. Possible
+	// values include INTERACTIVE and BATCH. The default value is
+	// INTERACTIVE.
+	Priority string `json:"priority,omitempty"`
+
+	// Query: [Required] SQL query text to execute. The useLegacySql field
+	// can be used to indicate whether the query uses legacy SQL or standard
+	// SQL.
+	Query string `json:"query,omitempty"`
+
+	// SchemaUpdateOptions: Allows the schema of the destination table to be
+	// updated as a side effect of the query job. Schema update options are
+	// supported in two cases: when writeDisposition is WRITE_APPEND; when
+	// writeDisposition is WRITE_TRUNCATE and the destination table is a
+	// partition of a table, specified by partition decorators. For normal
+	// tables, WRITE_TRUNCATE will always overwrite the schema. One or more
+	// of the following values are specified: ALLOW_FIELD_ADDITION: allow
+	// adding a nullable field to the schema. ALLOW_FIELD_RELAXATION: allow
+	// relaxing a required field in the original schema to nullable.
+	SchemaUpdateOptions []string `json:"schemaUpdateOptions,omitempty"`
+
+	// TableDefinitions: [Optional] If querying an external data source
+	// outside of BigQuery, describes the data format, location and other
+	// properties of the data source. By defining these properties, the data
+	// source can then be queried as if it were a standard BigQuery table.
+	TableDefinitions map[string]bigquery.ExternalDataConfiguration `json:"tableDefinitions,omitempty"`
+
+	// TimePartitioning: Time-based partitioning specification for the
+	// destination table. Only one of timePartitioning and rangePartitioning
+	// should be specified.
+	TimePartitioning *bigquery.TimePartitioning `json:"timePartitioning,omitempty"`
+
+	// UseLegacySQL: Specifies whether to use BigQuery's legacy SQL dialect
+	// for this query. The default value is true. If set to false, the query
+	// will use BigQuery's standard SQL:
+	// https://cloud.google.com/bigquery/sql-reference/ When useLegacySql is
+	// set to false, the value of flattenResults is ignored; query will be
+	// run as if flattenResults is false.
+	//
+	// Default: true
+	UseLegacySQL *bool `json:"useLegacySql,omitempty"`
+
+	// UseQueryCache: [Optional] Whether to look for the result in the query
+	// cache. The query cache is a best-effort cache that will be flushed
+	// whenever tables in the query are modified. Moreover, the query cache
+	// is only available when a query does not have a destination table
+	// specified. The default value is true.
+	//
+	// Default: true
+	UseQueryCache *bool `json:"useQueryCache,omitempty"`
```

Comment:
weird to have default true, with a pointer? default for golang is `false` for a bool. and with pointer its like a ternary value